src/cdk/v2/destinations/koala/procWorkflow.yaml

```diff
@@ -0,0 +1,69 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+
+steps:
+  - name: validateInput
+    template: |
+      $.assert(.message.type, "message Type is not present. Aborting message");
+      $.assert(.message.type in {{$.EventType.([.IDENTIFY, .TRACK])}}, 
+        "message type " + .message.type + " is not supported");
+      $.assertConfig(.destination.Config.publicKey, "publicKey is not present. Aborting message");
+      $.assert(.message.traits.email ?? .message.context.traits.email ?? .message.traits.ko_profile_id ?? .message.context.traits.ko_profile_id, "Neither email or ko_profile_id are present on traits. Aborting message");
+  - name: setMessageType
+    template: |
+      $.context.messageType = .message.type.toLowerCase();
+  - name: preparePayloadForIdentify
+    condition: $.context.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      const traits = .message.traits ?? .message.context.traits ?? {};
+      const koTraits = traits{~['ko_profile_id']}
+      const basePayload = {
+        email: .message.().({{{{$.getGenericPaths("email")}}}}),
+        profile_id: traits.ko_profile_id,
+        identifies: [{
+          type: $.context.messageType,
+          event: .message.event, 
+          sent_at: .message.().({{{{$.getGenericPaths("timestamp")}}}}),
+          traits: koTraits 
+        }]
+      };
+
+      $.context.payload = basePayload
+  - name: preparePayloadForTrack
+    condition: $.context.messageType === {{$.EventType.TRACK}}
+    template: |
+      const properties = .message.properties ??  {};
+      const basePayload = {
+        ip: .message.context.ip ?? .message.request_ip ?? , 
+        email: .message.().({{{{$.getGenericPaths("email")}}}}),
+        profile_id: .message.traits.ko_profile_id ?? .message.context.traits.ko_profile_id,
+        events: [{
+          type: $.context.messageType,
+          event: .message.event,
+          message_id: .message.messageId,
+          sent_at: .message.().({{{{$.getGenericPaths("timestamp")}}}}),
+          properties: properties,
+          context: .message.context
+        }]
+      };
+
+      $.context.payload = basePayload
+  - name: buildResponse
+    description: In batchMode we return payload directly
+    condition: $.batchMode
```

Comment:
Do we support batching? If yes what is the max batch size?

Reply:
Hy @Gauravudia Nice question, thanks for asking. The maiximum batch size its 30, we truncate  any events over 30

Reply:
In that case we have 2 options
- Implement the batch for identify, track call
   - ref sample destination that support batching: https://github.com/rudderlabs/rudder-transformer/blob/a7b30cccb4899ac917a7ac318f3fb22fa7f61094/src/cdk/v2/destinations/algolia/rtWorkflow.yaml#L34
- skip the batching for now and later we can add the batching support as part of enhancement.
    - remove the buildResponse step and keep buildResponseForProcessTransformation

Reply:
Just skip the batching step for now